tests/base/converters/test_dataframe_to_mds.py

```diff
@@ -0,0 +1,322 @@
+    def test_integration_conversion_local_and_remote(self, dataframe: Any, remote_local_dir: Any,
+                                                     merge_index: bool, keep_local: bool):
+        if not LOCAL_MANUAL_TEST:
+            pytest.skip('run local only. CI cluster does not have GCS service acct set up.')
+        out = remote_local_dir(cloud_prefix='gs://')
+        # bucket_name = 'mosaicml-composer-tests'
+        mds_kwargs = {
+            'out': out,
+            'columns': {
+                'id': 'str',
+                'dept': 'str'
+            },
+            'keep_local': keep_local,
+            'compression': 'zstd:7',
+            'hashes': ['sha1', 'xxh64'],
+            'size_limit': 1 << 26
+        }
+
+        mds_path, _ = dataframeToMDS(dataframe, merge_index=merge_index, mds_kwargs=mds_kwargs)
+
+        assert out == mds_path, f'returned mds_path: {mds_path} is not the same as out: {out}'
+
+        if keep_local == False:
+            assert (not os.path.exists(mds_path[0])), 'local folder were not removed'
+            return
+
+        assert (len(os.listdir(mds_path[0])) > 0), f'{mds_path[0]} is empty'
+        for d in os.listdir(mds_path[0]):
+            if os.path.isdir(os.path.join(mds_path[0], d)):
+                assert (os.path.exists(os.path.join(
+                    mds_path[0], d, 'index.json'))), f'No index.json found in subdirectory {d}'
+
+        if merge_index == True:
+            assert (os.path.exists(os.path.join(mds_path[0],
+                                                'index.json'))), 'No merged index found'
+        else:
+            assert not (
+                os.path.exists(os.path.join(mds_path[0], 'index.json'))
+            ), f'merged index is created at {mds_path[0]} when merge_index={merge_index} and keep_local={keep_local}'
+
+    @pytest.mark.usefixtures('remote_local_dir')
+    def test_integration_conversion_remote_only(self, dataframe: Any, remote_local_dir: Any):
+        if not LOCAL_MANUAL_TEST:
+            pytest.skip('run local only. CI cluster does not have GCS service acct set up.')
+        _, remote = remote_local_dir(cloud_prefix='gs://')
+        mds_kwargs = {
+            'out': remote,
+            'columns': {
+                'id': 'str',
+                'dept': 'str'
+            },
+        }
+
+        mds_path, _ = dataframeToMDS(dataframe, merge_index=True, mds_kwargs=mds_kwargs)
+
+        assert len(mds_path) == 2, 'returned mds is a str but should be a tuple (local, remote)'
+        assert not (os.path.exists(os.path.join(
+            mds_path[0], 'index.json'))), 'Local merged index was not removed successfully'
+        assert (len(os.listdir(mds_path[0])) > 0), f'{mds_path[0]} is not empty'
+
+    def test_simple_remote(self, dataframe: Any):
+        if not LOCAL_MANUAL_TEST:
+            pytest.skip('run local only. CI cluster does not have GCS service acct set up.')
+
+        out = 'gs://mosaicml-composer-tests/test_df2mds'
```

Comment:
Does this actually upload shard file to GCS ? I think unit test does not perform service authentication since they don't have credentials and it's expensive since the test runs for every commit.

Reply:
Yes, I have set LOCAL_MANUAL_TEST to be false. All the integration tests will skipped on CI nodes.